data/sql/updates/pending_db_world/scarlet_infantrymen.sql

```diff
@@ -0,0 +1,135 @@
+
+(12928200, 1, 2214.83, -5884.92, 100.855, NULL, 0, 0, 0, 100, 0),
+(12928200, 2, 2209.52, -5885, 101.008, NULL, 0, 0, 0, 100, 0),
+(12928200, 3, 2204.1, -5883.88, 101.159, NULL, 0, 0, 0, 100, 0),
+(12928200, 4, 2195.65, -5882.51, 101.208, NULL, 0, 0, 0, 100, 0),
+(12928200, 5, 2187.06, -5880.64, 101.065, NULL, 0, 0, 0, 100, 0),
+(12928200, 6, 2195.65, -5882.51, 101.208, NULL, 0, 0, 0, 100, 0),
+(12928200, 7, 2204.1, -5883.88, 101.159, NULL, 0, 0, 0, 100, 0),
+(12928200, 8, 2209.52, -5885, 101.008, NULL, 0, 0, 0, 100, 0),
+(12928200, 9, 2214.83, -5884.92, 100.855, NULL, 0, 0, 0, 100, 0),
+(12928200, 10, 2221.08, -5885.01, 100.68, NULL, 0, 0, 0, 100, 0),
+
+(12928300, 1, 2358.94, -5843.76, 103.624, NULL, 0, 0, 0, 100, 0),
+(12928300, 2, 2367.94, -5842.23, 106.039, NULL, 0, 0, 0, 100, 0),
+(12928300, 3, 2376.98, -5841.91, 108.059, NULL, 0, 0, 0, 100, 0),
+(12928300, 4, 2385.39, -5843.93, 108.588, NULL, 0, 0, 0, 100, 0),
+(12928300, 5, 2392.12, -5844.45, 108.994, NULL, 0, 0, 0, 100, 0),
+(12928300, 6, 2385.39, -5843.93, 108.588, NULL, 0, 0, 0, 100, 0),
+(12928300, 7, 2376.98, -5841.91, 108.059, NULL, 0, 0, 0, 100, 0),
+(12928300, 8, 2367.94, -5842.23, 106.039, NULL, 0, 0, 0, 100, 0),
+(12928300, 9, 2358.94, -5843.76, 103.624, NULL, 0, 0, 0, 100, 0),
+(12928300, 10, 2353.2, -5844.39, 101.593, NULL, 0, 0, 0, 100, 0),
+
+(12928400, 1, 2385.39, -5843.93, 108.588, NULL, 0, 0, 0, 100, 0),
+(12928400, 2, 2376.98, -5841.91, 108.059, NULL, 0, 0, 0, 100, 0),
+(12928400, 3, 2367.94, -5842.23, 106.039, NULL, 0, 0, 0, 100, 0),
+(12928400, 4, 2358.94, -5843.76, 103.624, NULL, 0, 0, 0, 100, 0),
+(12928400, 5, 2353.2, -5844.39, 101.593, NULL, 0, 0, 0, 100, 0),
+(12928400, 6, 2358.94, -5843.76, 103.624, NULL, 0, 0, 0, 100, 0),
+(12928400, 7, 2367.94, -5842.23, 106.039, NULL, 0, 0, 0, 100, 0),
+(12928400, 8, 2376.98, -5841.91, 108.059, NULL, 0, 0, 0, 100, 0),
+(12928400, 9, 2385.39, -5843.93, 108.588, NULL, 0, 0, 0, 100, 0),
+(12928400, 10, 2392.12, -5844.45, 108.994, NULL, 0, 0, 0, 100, 0);
+
+
+-- Change movement type and wander distance
+
+UPDATE `creature` SET `wander_distance` = 0, `MovementType` = 2 WHERE `guid` = 129272 AND `id1` = 28609;
+UPDATE `creature` SET `wander_distance` = 0, `MovementType` = 2 WHERE `guid` = 129282 AND `id1` = 28609;
+UPDATE `creature` SET `wander_distance` = 0, `MovementType` = 2 WHERE `guid` = 129283 AND `id1` = 28609;
+UPDATE `creature` SET `wander_distance` = 0, `MovementType` = 2 WHERE `guid` = 129284 AND `id1` = 28609;
+
+
+-- Set path id
+
+UPDATE `creature_addon` SET `path_id` = 12927200 WHERE (`guid` IN (129272));
+UPDATE `creature_addon` SET `path_id` = 12928200 WHERE (`guid` IN (129282));
+UPDATE `creature_addon` SET `path_id` = 12928300 WHERE (`guid` IN (129283));
+UPDATE `creature_addon` SET `path_id` = 12928400 WHERE (`guid` IN (129284));
+
+
+-- Set Stealth Movement (already present in one of these Guids)
+
+UPDATE `creature_addon` SET `bytes1` = 131072 WHERE (`guid` IN (129272, 129282, 129283, 129284));
```

Comment:
Not sure about this, invisibility/stealth is usually handled through an aura. A sniffs of these creatures would show the aura quite clearly.

Reply:
I already thought about the aura option, but I'm not good at sniffing and I've tried many times to use a stealth aura, but none of the ones I've used perfectly replicate their behaviour (stealth auras usually have a sound effect, for example). If anyone can tell me if these npc's have an aura (and the aura id), I'll be happy to modify the fix. 
If I can be honest, I am not satisfied with this method either.

Reply:
I've spotted [52103](https://wowgaming.altervista.org/aowow/?spell=52103) on a [Scarlet Infantryman](https://wowgaming.altervista.org/aowow/?npc=28609) in a 3.4.2 50664 sniff, looks like the one you're after.

Reply:
> I've spotted [52103](https://wowgaming.altervista.org/aowow/?spell=52103) on a [Scarlet Infantryman](https://wowgaming.altervista.org/aowow/?npc=28609) in a 3.4.2 50664 sniff, looks like the one you're after.

One day I will have to learn how to read the sniffs xD. 
It is correct, thank you bench :)

Reply:
> One day I will have to learn how to read the sniffs

The way I found this one was to search a sniff that contained a creature with the aura for the following string: `Entry: 28609`. This yields every message that involves a Scarlet Infantryman. Specifically for this issue, I then scrolled through every instance with particular attention paid to `SMSG_SPELL_START` and especially `SMSG_AURA_UPDATE`. In this case, I found the spell in the aura update, which just is an update message for the auras currently on a unit.

It takes practice, but most of the time, it's not as complex as it seems. 😄

Reply:
In fact I didn't see it 😄
![aura](https://github.com/user-attachments/assets/e1d91394-a208-4225-88b8-a1696241d906)

Reply:
Question: do these creatures still function as expected if this `bytes1` value isn't changed?

Reply:
With the aura, they don't need that byte anymore. It can be setted to 0.